homeassistant/components/climaveneta_imxw/climate.py

```diff
@@ -0,0 +1,339 @@
+"""Support for the Mitsubishi-Climaveneta iMXW fancoil series."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    PLATFORM_SCHEMA,
+    SWING_OFF,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.components.modbus import (
+    CALL_TYPE_REGISTER_HOLDING,
+    CALL_TYPE_WRITE_REGISTER,
+    CONF_HUB,
+    DEFAULT_HUB,
+    ModbusHub,
+    get_hub,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_NAME,
+    CONF_SLAVE,
+    DEVICE_DEFAULT_NAME,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_HUB, default=DEFAULT_HUB): cv.string,
+        vol.Required(CONF_SLAVE): vol.All(int, vol.Range(min=0, max=255)),
+        vol.Optional(CONF_NAME, default=DEVICE_DEFAULT_NAME): cv.string,
+    }
+)
```

Comment:
We no longer allow integrations to add or change a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Please note that this integration connects to a device or service and another Architecture Decision Record applies that disallows the use of YAML configuration in favor of a configuration flow via the UI:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

As these changes often involve a bit of work and some significant shift in the current code, we will close this PR for now.

Thanks already! :+1: